src/orchard/classpath.clj

```diff
@@ -88,6 +89,28 @@
 
 ;;; Boot's hack - previously part of cider-nrepl
 
+(defn boot-classloader
+  "Creates a class-loader that knows original source files paths in Boot project."
+  []
+  (let [class-path (System/getProperty "fake.class.path")
+        dir-separator (System/getProperty "file.separator")
+        paths (str/split class-path (re-pattern (System/getProperty "path.separator")))
+        urls (map
+              (fn [path]
+                (let [url (if (re-find #".jar$" path)
+                            (str "file:" path)
+                            (str "file:" path dir-separator))]
+                  (new java.net.URL url)))
+              paths)]
+    ;; TODO: Figure out how to add the JDK sources here
+    (new java.net.URLClassLoader (into-array java.net.URL urls))))
+
+(defn boot-hack-classloader
+  []
+  (if (u/boot-project?)
+    (boot-classloader)
+    (.getContextClassLoader (Thread/currentThread))))
```

Comment:
There's a `context-classloader` helper that does this. I wonder if we shouldn't modify it instead, although I'm not certain if we shouldn't modify it instead.

Reply:
Maybe yeah I can call that function to avoid future confusion

Reply:
I was also a bit scared about that TODO 😀